app/routes/vet/reference.js

```diff
@@ -2,10 +2,13 @@ const Joi = require('joi')
 const session = require('../../session')
 const { vetSignup: { reference: referenceKey } } = require('../../session/keys')
 const { reference: referenceErrorMessages } = require('../../../app/lib/error-messages')
+const { applicationRequestQueue, fetchApplicationRequestMsgType, applicationResponseQueue } = require('../../config')
+const { sendMessage, receiveMessage } = require('../../messaging')
 
 // TODO: implement proper application lookup
-function getApplication (reference) {
-  return true
+function getApplication (reference, sessionId) {
+  sendMessage({ application: reference, sessionId }, fetchApplicationRequestMsgType, applicationRequestQueue, { sessionId })
```

Comment:
This is a potential future refactor to reflect the business language of `bookingReference` (or similar).

Currently this differs from the response object as that uses `applicationId` rather than just `application`. It might be worth making those consistent now to ease the refactor.